src/services/dom/parser.js

```diff
@@ -58,7 +58,7 @@ export class Parser {
     data: ?FormData,
     httpMethod: ?HttpMethod,
     acceptContentType: string = CONTENT_TYPE.APPLICATION_VND_HYPERVIEW_XML,
-  ): Promise<Document> => {
+  ): Promise<{ doc: Document, isStale: boolean }> => {
```

Comment:
just thinking ahead a bit, how will we distinguish between stale due to error vs stale while revalidating? Maybe instead of a boolean, should we return an enum?

Reply:
addressed in https://github.com/Instawork/hyperview/pull/484/commits/664aefe39ad93cc1f0085b0e14ab5175a65e39bb